test.js

```diff
@@ -1,17 +1,49 @@
 import test from 'ava';
 import sortKeys from '.';
 
+function deepEqualInOrder(t, actual, expected) {
+	t.deepEqual(actual, expected);
+
+	const seen = new Set();
+
+	function assertSameKeysInOrder(object1, object2) {
+		// This function assumes the objects given are already deep equal.
+
+		if (seen.has(object1) && seen.has(object2)) {
+			return;
+		}
+
+		seen.add(object1);
+		seen.add(object2);
+
+		if (Array.isArray(object1)) {
+			for (let i = 0; i < object1.length; i++) {
```

Comment:
You can use `Object.entries` here and a for-of loop.

Reply:
Hmm, indeed I can, but personally I think that would be less readable here. In particular, [`Object.entries` is documented to return **string-keyed** props of **objects**](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries), and the fact that it works on arrays is probably an obscure consequence of the early days of JavaScript.

```js
Object.entries([10, 20]);
//=> [['0', 10], ['1', 20]]
```

And it returns indexes as strings, which I don't like.

Can I leave it this way? If not, I can

Reply:
Hmm, I guess you intended to say `Array.prototype.entries`. That would make more sense.

I still prefer the way I wrote it, but I don't mind switching in this case.

Reply:
I ended up using `Array.prototype.keys` instead. What do you think? 13ae015

Reply:
Yeah, looks good. I'm just staying away from normal loops as they often have the chance of off-by-one errors.